### PR TITLE
feat: improve proposal billing display

### LINKED ISSUE #2267: [cms] Show monthly subtotals for Proposals Owned
When reviewing billing against my proposal it would help to see a subtotal (sum of labor and expenses) for each month. Some proposals promise a monthly expense limit and this feature is needed to facilitate that.

_extracted from https://github.com/decred/politeia/issues/1328#issuecomment-731605967 _

### PR DESCRIPTION
**Pull request by tiagoalvesdulce**
Closes #2267 

### UI Changes Screenshot

Screen Shot 2020-12-29 at 17 12 42

I think it's looking pretty decent now. Suggestions on how to improve this page are appreciated.

### CODE REVIEW

**Comment by xaur** [comment]
Looking good!

One small idea is to show year for each month, e.g. `March` -> `March 2020` to give each month's section full context (e.g. imagine that a screenshot cut of March's table will be more self-contained with a year on it) and be consistent with invoice submit and review pages. And then the "Billings xxx" grouping becomes unnecessary.

Just an idea, not blocking this PR.

File: src/containers/User/Detail/ProposalsOwned/ProposalBilling.jsx [hunk 1]
```diff
@@ -17,40 +27,79 @@ const headers = [
...
-        Subdomain: subdomain,
-        Description: description,
-        Labor: usdFormatter.format(labor),
-        Expenses: usdFormatter.format(expenses),
-        Total: usdFormatter.format(labor + expenses)
-      };
-    }
-  );
+  const groupedByYear = groupBy((el) => {
+    console.log(el);
```
Comment (amassarwi) [inline src/containers/User/Detail/ProposalsOwned/ProposalBilling.jsx:95]: console

File: src/containers/User/Detail/ProposalsOwned/ProposalBilling.jsx [hunk 2]
```diff
@@ -17,40 +27,79 @@ const headers = [
...
+    <>
+      <Table headers={headers} data={data} linesPerPage={50} />
+      <H3 className={styles.totalText}>Total: {usdFormatter.format(total)}</H3>
+    </>
+  );
+};
+
+const YearItems = ({ year, yearItems }) => {
+  const groupedByMonth = groupBy((el) => {
+    console.log(el);
```
Comment (amassarwi) [inline src/containers/User/Detail/ProposalsOwned/ProposalBilling.jsx:68]: log

File: src/constants.js [hunk 3]
```diff
@@ -164,3 +164,18 @@ export const MONTHS_LABELS = [
   "Nov",
   "Dec"
 ];
+
+export const FULL_MONTHS = [
```
Comment (amassarwi) [inline src/constants.js:168]: We have another months list here:

https://github.com/decred/politeiagui/blob/master/src/utils.js#L1

I guess we could just have one in constants and use it everywhere.

**Review by amassarwi — APPROVED** [review]
Looks good to me, left some minor comments.

**Pull request merged**